Remove writing pin/block records from social #905

boostmultifruit opened the pull request:
Resolve #905.
What is done:
See below: https://github.com/cyberway/golos.contracts/pull/906#issuecomment-536137190

gropox commented:
А что будет вместо удаленного pinblock-а? А то я начал использовать в своих ботах эти данные. Есть уже план? 

Спасибо

boostmultifruit commented:
@gropox какие именно данные? Сами экшны pin, unpin, block, unblock остаются, просто больше не хранятся записи в монге, теперь их хранить будет IO.

gropox commented:
@boostmultifruit вот как раз данные из таблицы pinblock и брал. 

В смартконтракте бота 

```
		tables::pinblock_table blacklist_table(golos::config::social_name, get_self().value);
		auto block = blacklist_table.find(author.value);
		if(block != blacklist_table.end() && block->blocking) {
			return "Автор в черном списке @" + author.to_string();			
		}

```

и в другом телеграмном боте, из базы монги вытаскивал черный список, что бы не информировать людей о сообщениях от заблокированных пользователей.

Это было очень удобно. Если уберете, то окологолосинным приложениям придется туго и тоже надо будет ловить экшены и хранить все в своей базе. 

Со смартом ладно, там я могу свой черный список завести, а вот с ботом чуть тяжелее. Свою базу делать + сканировать эвенты с самого начала, если ботом к блокчейну не сначала подключаться.

afalaleev commented:
At the current time, ban and pin lists doesn't contain all exist records.
As result client fails on removing pin or ban records.

So, the main change is to disable of fails on removing not-existing records in contacts tables.

And as @gropox says, we should keep saving of pin/block records.

afalaleev commented on genesis/genesis-info.json.tmpl:
```diff
@@ -367,7 +367,6 @@
                 "cashout_window": {"window":604800, "upvote_lockout":60},
                 "max_beneficiaries": {"value":64},
                 "max_comment_depth": {"value":127},
-                "social_acc": {"value":"gls.social"},
```
You should not remove it from genesis.

afalaleev commented on golos.social/golos.social.abi:
```diff
@@ -87,23 +78,10 @@
   ],
   "events": [ ],
   "tables": [
-     {
```
Applying of this ABI will fail, because table has records.
Only empty tables can be removed.

afalaleev commented on scripts/golos-boot-sequence/golos-boot-sequence.py:
```diff
@@ -398,7 +398,6 @@ def initCommunity():
             ['st_cashout_window', {'window': 7*24*3600, 'upvote_lockout': 60}],
             ['st_max_beneficiaries', {'value': 64}],
             ['st_max_comment_depth', {'value': 127}],
-            ['st_social_acc', {'value': 'gls.social'}],
```
You shouldn't change genesis, because its in history now.

afalaleev commented on golos.publication/golos.publication.cpp:
```diff
@@ -160,13 +154,6 @@ void publication::create_message(
         curators_prcnt = curators_prcnt_param.min_curators_prcnt;
     }
 
-    if (parent_id.author) {
```
This removing  is OK, because block list doesn't contain all records.
It can be restored later, after syncing of pin/block lists with ee-genesis.

boostmultifruit commented:
@afalaleev What is done:
- Allowed to call `unpin` and `unblock` without pin/block record. (For this case to these actions added check that pinning/blocking account exists.)
- Removed `record_is_empty()` check which was added by AlexandrDedckov and which was totally wrong, because account cannot be pinned and blocked simultaneously.
- Updated tests.

s-medvedev commented on golos.social/golos.social.cpp:
```diff
@@ -15,16 +15,9 @@ void social::pin(name pinner, name pinning) {
+    if (itr != table.end()) {
         eosio::check(!itr->blocking, "You have blocked this account. Unblock it before pinning");
         eosio::check(!itr->pinning, "You already have pinned this account");
-
```
What will happen if two checks are successful? I understand that this is not possible with correct entries. But if there is a record in the database, then the action will be performed, but no effect will occur. It is worth in this case at least to assert (such a situation is impossible, an error in the data).
And it turns out that the `pin`/`block` behavior is different in this case.

boostmultifruit replied:
@s-medvedev Added `SYSTEM` `eosio::check`s.
